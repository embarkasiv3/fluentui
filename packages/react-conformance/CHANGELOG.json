{
  "name": "@fluentui/react-conformance",
  "entries": [
    {
      "date": "Thu, 18 Feb 2021 12:27:34 GMT",
      "tag": "@fluentui/react-conformance_v1.0.0",
      "version": "1.0.0",
      "comments": {
        "prerelease": [
          {
            "comment": "Bump @fluentui/eslint-plugin to v1.0.0-beta.2",
            "author": "elcraig@microsoft.com",
            "commit": "1072765ed2e0ffda34688d84d376a8bc4bf928f0",
            "package": "@fluentui/react-conformance"
          }
        ]
      }
    },
    {
      "date": "Thu, 11 Feb 2021 12:18:41 GMT",
      "tag": "@fluentui/react-conformance_v1.0.0",
      "version": "1.0.0",
      "comments": {
        "none": [
          {
            "comment": "Bump @fluentui/eslint-plugin to v1.0.0-beta.1",
            "author": "martinhochel@microsoft.com",
            "commit": "7566015a7edd355b4fcd3796bc8f44f732ef0877",
            "package": "@fluentui/react-conformance"
          }
        ]
      }
    }
  ]
}
