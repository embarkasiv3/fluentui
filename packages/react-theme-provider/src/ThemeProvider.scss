/**
 * The ThemeProvider has a root consuming class which will apply the current
 * token values to the container DIV, so that fonts and default colors, which
 * refer to token values, can be applied.
 */
.root {
  background: var(--body-fill);
  color: var(--body-text);
  font-family: var(--body-fontFamily);
  font-weight: var(--body-fontWeight);
  font-size: var(--body-fontSize);
  line-height: var(--body-fontLineHeight);

  > * {
    box-sizing: border-box;
  }
}
