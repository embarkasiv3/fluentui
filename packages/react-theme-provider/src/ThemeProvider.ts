export * from './components/ThemeProvider/index';
