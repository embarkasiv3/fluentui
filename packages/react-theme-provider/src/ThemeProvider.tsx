import { useMergedRefs } from '@fluentui/react-hooks';
import { mergeThemes, themeToCSSVariables, PartialTheme, Theme } from '@fluentui/react-theme';
import { getSlots, makeMergeProps } from '@fluentui/react-utils';
import * as React from 'react';

import { internal__ThemeContext, useTheme } from './context';

export interface ThemeProviderProps extends React.HTMLAttributes<HTMLElement> {
  theme?: PartialTheme | Theme;
}
export interface ThemeProviderState extends React.HTMLAttributes<HTMLElement> {
  theme: Theme;
}

const mergeProps = makeMergeProps<ThemeProviderState>();

export function useThemeProviderState(draftState: ThemeProviderState) {
  const parentTheme = useTheme();
  const localTheme = draftState.theme;

  draftState.theme = mergeThemes(parentTheme, localTheme);
  draftState.style = React.useMemo(() => {
    // TODO: should we consider insertion to head?
    //       - how to modify, remove styles?
    //       - SSR rendering

    // TODO: what variables should be rendered? Merged or only changed?
    // TODO: how we will proceed with Portals?
    return {
      ...draftState.style,
      ...themeToCSSVariables(draftState.theme),
    };
  }, [draftState.style, draftState.theme]);
}

export function renderThemeProvider(state: ThemeProviderState) {
  const { slots, slotProps } = getSlots(state);
  const { theme } = state;

  return (
    <internal__ThemeContext.Provider value={theme}>
      <slots.root {...slotProps.root} />
    </internal__ThemeContext.Provider>
  );
}

/**
 * Returns the ThemeProvider render function and calculated state, given user input, ref, and
 * a set of default prop values.
 */
export function useThemeProvider(props: ThemeProviderProps, ref: React.Ref<HTMLElement>) {
  const rootRef = useMergedRefs(ref, React.useRef<HTMLElement>(null));
  const state = mergeProps(
    {
      ref: rootRef,
      as: 'div',
    },
    {},
    props,
  );

  useThemeProviderState(state);

  return {
    state,
    render: renderThemeProvider,
  };
}

/**
 * Used to provide CSS variables to DOM and theme tokens via React Context.
 */
export const ThemeProvider: React.FunctionComponent<ThemeProviderProps> = React.forwardRef<
  HTMLDivElement,
  ThemeProviderProps
>((props: ThemeProviderProps, ref: React.Ref<HTMLDivElement>) => {
  const { render, state } = useThemeProvider(props, ref);

  return render(state);
});

ThemeProvider.displayName = 'ThemeProvider';
