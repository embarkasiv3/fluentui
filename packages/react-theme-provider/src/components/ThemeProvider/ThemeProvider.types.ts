import * as React from 'react';
import { ComponentProps } from '@fluentui/react-utilities';
import { PartialTheme, Theme } from '@fluentui/react-theme';

export interface ThemeProviderProps extends ComponentProps, React.HTMLAttributes<HTMLElement> {
  theme?: PartialTheme | Theme;
}

export interface ThemeProviderState extends ThemeProviderProps {
  /**
   * Ref to the root slot
   */
  ref: React.MutableRefObject<HTMLElement>;
  theme: Theme;
}
