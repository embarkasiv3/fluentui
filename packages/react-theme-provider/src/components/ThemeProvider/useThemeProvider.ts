import * as React from 'react';
import { makeMergePropsCompat, resolveShorthandProps, useMergedRefs } from '@fluentui/react-utilities';
import { ThemeProviderProps, ThemeProviderState } from './ThemeProvider.types';
import { useTheme } from '@fluentui/react-shared-contexts';
import { mergeThemes, themeToCSSVariables } from '@fluentui/react-theme';

export const themeProviderShorthandProps: (keyof ThemeProviderProps)[] = [];

// eslint-disable-next-line deprecation/deprecation
const mergeProps = makeMergePropsCompat<ThemeProviderState>({ deepMerge: themeProviderShorthandProps });

/**
 * Create the state required to render ThemeProvider.
 *
 * The returned state can be modified with hooks such as useThemeProviderStyles,
 * before being passed to renderThemeProvider.
 *
 * @param props - props from this instance of ThemeProvider
 * @param ref - reference to root HTMLElement of ThemeProvider
 * @param defaultProps - (optional) default prop values provided by the implementing type
 */
export const useThemeProvider = (
  props: ThemeProviderProps,
  ref: React.Ref<HTMLElement>,
  defaultProps?: ThemeProviderProps,
): ThemeProviderState => {
  const state = mergeProps(
    {
      ref: useMergedRefs(ref, React.useRef(null)),
      as: 'div',
    },
    defaultProps,
    resolveShorthandProps(props, themeProviderShorthandProps),
  );

  const parentTheme = useTheme();
  const localTheme = state.theme;

  state.theme = mergeThemes(parentTheme, localTheme);
  state.style = React.useMemo(() => {
    // TODO: should we consider insertion to head?
    //       - how to modify, remove styles?
    //       - SSR rendering

    // TODO: what variables should be rendered? Merged or only changed?
    // TODO: how we will proceed with Portals?
    return {
      ...state.style,
      ...themeToCSSVariables(state.theme),
    };
  }, [state.style, state.theme]);

  return state;
};
