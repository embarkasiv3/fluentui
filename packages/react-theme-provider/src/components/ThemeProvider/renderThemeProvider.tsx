import * as React from 'react';
import { getSlots } from '@fluentui/react-utilities';
import { ThemeProviderState } from './ThemeProvider.types';
import { themeProviderShorthandProps } from './useThemeProvider';
import { ThemeContext } from '@fluentui/react-shared-contexts';

/**
 * Render the final JSX of ThemeProvider
 */
export const renderThemeProvider = (state: ThemeProviderState) => {
  const { slots, slotProps } = getSlots(state, themeProviderShorthandProps);

  return (
    <ThemeContext.Provider value={state.theme}>
      <slots.root {...slotProps.root}>{state.children}</slots.root>
    </ThemeContext.Provider>
  );
};
