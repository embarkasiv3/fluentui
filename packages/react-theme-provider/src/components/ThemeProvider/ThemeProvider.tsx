import * as React from 'react';
import { useThemeProvider } from './useThemeProvider';
import { ThemeProviderProps } from './ThemeProvider.types';
import { renderThemeProvider } from './renderThemeProvider';

export const ThemeProvider = React.forwardRef<HTMLElement, ThemeProviderProps>((props, ref) => {
  const state = useThemeProvider(props, ref);
  return renderThemeProvider(state);
});

ThemeProvider.displayName = 'ThemeProvider';
