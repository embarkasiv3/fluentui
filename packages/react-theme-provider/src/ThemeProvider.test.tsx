import { isConformant } from '@fluentui/react-conformance';
import * as path from 'path';

import { ThemeProvider } from './ThemeProvider';

describe('ThemeProvider (isConformant)', () => {
  isConformant({
    Component: ThemeProvider,

    componentPath: path.resolve(__dirname, 'ThemeProvider.tsx'),
    displayName: 'ThemeProvider',
  });
});
