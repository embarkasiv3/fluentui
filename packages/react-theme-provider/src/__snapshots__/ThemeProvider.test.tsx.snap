// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider can apply body theme to body 1`] = `
<body
  class=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: black;
        --color-body-contentColor: white;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: var(--color-brand-disabled-background);
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: var(--color-brand-disabled-contentColor);
        --color-brand-disabled-secondaryContentColor: var(--color-brand-disabled-contentColor);
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: var(--color-brand-background);
        --color-brand-focused-borderColor: var(--color-brand-borderColor);
        --color-brand-focused-contentColor: var(--color-brand-contentColor);
        --color-brand-focused-iconColor: var(--color-brand-iconColor);
        --color-brand-focused-secondaryContentColor: var(--color-brand-focused-contentColor);
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: var(--color-brand-borderColor);
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: var(--color-brand-borderColor);
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: var(--color-brand-iconColor);
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
      }
      {
        -moz-osx-font-smoothing: var(--body-mozOsxFontSmoothing);
        -webkit-font-smoothing: var(--body-webkitFontSmoothing);
        background: var(--color-body-background);
        color: var(--color-body-contentColor);
        font-family: var(--body-fontFamily);
        font-size: var(--body-fontSize);
        font-weight: var(--body-fontWeight);
      }
/>
`;

exports[`ThemeProvider can apply body theme to body 2`] = `
<div
  className=
      foo
      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: black;
        --color-body-contentColor: white;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: var(--color-brand-disabled-background);
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: var(--color-brand-disabled-contentColor);
        --color-brand-disabled-secondaryContentColor: var(--color-brand-disabled-contentColor);
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: var(--color-brand-background);
        --color-brand-focused-borderColor: var(--color-brand-borderColor);
        --color-brand-focused-contentColor: var(--color-brand-contentColor);
        --color-brand-focused-iconColor: var(--color-brand-iconColor);
        --color-brand-focused-secondaryContentColor: var(--color-brand-focused-contentColor);
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: var(--color-brand-borderColor);
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: var(--color-brand-borderColor);
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: var(--color-brand-iconColor);
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
      }
>
  app
</div>
`;

exports[`ThemeProvider can apply body theme to none 1`] = `
<div
  className=
      foo
      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: black;
        --color-body-contentColor: white;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: var(--color-brand-disabled-background);
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: var(--color-brand-disabled-contentColor);
        --color-brand-disabled-secondaryContentColor: var(--color-brand-disabled-contentColor);
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: var(--color-brand-background);
        --color-brand-focused-borderColor: var(--color-brand-borderColor);
        --color-brand-focused-contentColor: var(--color-brand-contentColor);
        --color-brand-focused-iconColor: var(--color-brand-iconColor);
        --color-brand-focused-secondaryContentColor: var(--color-brand-focused-contentColor);
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: var(--color-brand-borderColor);
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: var(--color-brand-borderColor);
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: var(--color-brand-iconColor);
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
      }
>
  app
</div>
`;

exports[`ThemeProvider can handle a partial theme 1`] = `
<div
  className=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: #ffffff;
        --color-body-contentColor: #323130;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: var(--color-brand-disabled-background);
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: var(--color-brand-disabled-contentColor);
        --color-brand-disabled-secondaryContentColor: var(--color-brand-disabled-contentColor);
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: var(--color-brand-background);
        --color-brand-focused-borderColor: var(--color-brand-borderColor);
        --color-brand-focused-contentColor: var(--color-brand-contentColor);
        --color-brand-focused-iconColor: var(--color-brand-iconColor);
        --color-brand-focused-secondaryContentColor: var(--color-brand-focused-contentColor);
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: var(--color-brand-borderColor);
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: var(--color-brand-borderColor);
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: var(--color-brand-iconColor);
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
        --foo-background: red;
      }
      {
        -moz-osx-font-smoothing: var(--body-mozOsxFontSmoothing);
        -webkit-font-smoothing: var(--body-webkitFontSmoothing);
        background: var(--color-body-background);
        color: var(--color-body-contentColor);
        font-family: var(--body-fontFamily);
        font-size: var(--body-fontSize);
        font-weight: var(--body-fontWeight);
      }
>
  Hello
</div>
`;

exports[`ThemeProvider renders a div 1`] = `
<div
  className=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: #ffffff;
        --color-body-contentColor: #323130;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: var(--color-brand-disabled-background);
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: var(--color-brand-disabled-contentColor);
        --color-brand-disabled-secondaryContentColor: var(--color-brand-disabled-contentColor);
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: var(--color-brand-background);
        --color-brand-focused-borderColor: var(--color-brand-borderColor);
        --color-brand-focused-contentColor: var(--color-brand-contentColor);
        --color-brand-focused-iconColor: var(--color-brand-iconColor);
        --color-brand-focused-secondaryContentColor: var(--color-brand-focused-contentColor);
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: var(--color-brand-borderColor);
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: var(--color-brand-borderColor);
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: var(--color-brand-iconColor);
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
      }
      {
        -moz-osx-font-smoothing: var(--body-mozOsxFontSmoothing);
        -webkit-font-smoothing: var(--body-webkitFontSmoothing);
        background: var(--color-body-background);
        color: var(--color-body-contentColor);
        font-family: var(--body-fontFamily);
        font-size: var(--body-fontSize);
        font-weight: var(--body-fontWeight);
      }
>
  Hello
</div>
`;

exports[`ThemeProvider renders a div with styling 1`] = `
<div
  className=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: white;
        --color-body-contentColor: black;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: var(--color-brand-disabled-background);
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: var(--color-brand-disabled-contentColor);
        --color-brand-disabled-secondaryContentColor: var(--color-brand-disabled-contentColor);
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: var(--color-brand-background);
        --color-brand-focused-borderColor: var(--color-brand-borderColor);
        --color-brand-focused-contentColor: var(--color-brand-contentColor);
        --color-brand-focused-iconColor: var(--color-brand-iconColor);
        --color-brand-focused-secondaryContentColor: var(--color-brand-focused-contentColor);
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: var(--color-brand-borderColor);
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: var(--color-brand-borderColor);
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: var(--color-brand-iconColor);
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
      }
      {
        -moz-osx-font-smoothing: var(--body-mozOsxFontSmoothing);
        -webkit-font-smoothing: var(--body-webkitFontSmoothing);
        background: var(--color-body-background);
        color: var(--color-body-contentColor);
        font-family: var(--body-fontFamily);
        font-size: var(--body-fontSize);
        font-weight: var(--body-fontWeight);
      }
>
  Hello
</div>
`;

exports[`ThemeProvider renders nested themes 1`] = `
<div
  className=

      {
        --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
        --body-fontSize: 14px;
        --body-fontWeight: 400;
        --body-mozOsxFontSmoothing: grayscale;
        --body-webkitFontSmoothing: antialiased;
        --color-body-background: white;
        --color-body-contentColor: black;
        --color-brand-background: #0078d4;
        --color-brand-borderColor: transparent;
        --color-brand-checked-background: #005a9e;
        --color-brand-checked-contentColor: #ffffff;
        --color-brand-checkedHovered-background: #005a9e;
        --color-brand-checkedHovered-contentColor: #ffffff;
        --color-brand-contentColor: #ffffff;
        --color-brand-disabled-background: #f3f2f1;
        --color-brand-disabled-borderColor: var(--color-brand-disabled-background);
        --color-brand-disabled-contentColor: #a19f9d;
        --color-brand-disabled-dividerColor: #c8c6c4;
        --color-brand-disabled-iconColor: var(--color-brand-disabled-contentColor);
        --color-brand-disabled-secondaryContentColor: var(--color-brand-disabled-contentColor);
        --color-brand-dividerColor: #ffffff;
        --color-brand-focused-background: var(--color-brand-background);
        --color-brand-focused-borderColor: var(--color-brand-borderColor);
        --color-brand-focused-contentColor: var(--color-brand-contentColor);
        --color-brand-focused-iconColor: var(--color-brand-iconColor);
        --color-brand-focused-secondaryContentColor: var(--color-brand-focused-contentColor);
        --color-brand-hovered-background: #106ebe;
        --color-brand-hovered-borderColor: var(--color-brand-borderColor);
        --color-brand-hovered-contentColor: #ffffff;
        --color-brand-hovered-secondaryContentColor: #ffffff;
        --color-brand-iconColor: #ffffff;
        --color-brand-pressed-background: #005a9e;
        --color-brand-pressed-borderColor: var(--color-brand-borderColor);
        --color-brand-pressed-contentColor: #ffffff;
        --color-brand-pressed-iconColor: var(--color-brand-iconColor);
        --color-brand-pressed-secondaryContentColor: #ffffff;
        --color-brand-secondaryContentColor: #ffffff;
      }
      {
        -moz-osx-font-smoothing: var(--body-mozOsxFontSmoothing);
        -webkit-font-smoothing: var(--body-webkitFontSmoothing);
        background: var(--color-body-background);
        color: var(--color-body-contentColor);
        font-family: var(--body-fontFamily);
        font-size: var(--body-fontSize);
        font-weight: var(--body-fontWeight);
      }
>
  <div>
    Light theme
  </div>
  <div
    className=

        {
          --body-fontFamily: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
          --body-fontSize: 14px;
          --body-fontWeight: 400;
          --body-mozOsxFontSmoothing: grayscale;
          --body-webkitFontSmoothing: antialiased;
          --color-body-background: black;
          --color-body-contentColor: white;
          --color-brand-background: #0078d4;
          --color-brand-borderColor: transparent;
          --color-brand-checked-background: #005a9e;
          --color-brand-checked-contentColor: #ffffff;
          --color-brand-checkedHovered-background: #005a9e;
          --color-brand-checkedHovered-contentColor: #ffffff;
          --color-brand-contentColor: #ffffff;
          --color-brand-disabled-background: #f3f2f1;
          --color-brand-disabled-borderColor: var(--color-brand-disabled-background);
          --color-brand-disabled-contentColor: #a19f9d;
          --color-brand-disabled-dividerColor: #c8c6c4;
          --color-brand-disabled-iconColor: var(--color-brand-disabled-contentColor);
          --color-brand-disabled-secondaryContentColor: var(--color-brand-disabled-contentColor);
          --color-brand-dividerColor: #ffffff;
          --color-brand-focused-background: var(--color-brand-background);
          --color-brand-focused-borderColor: var(--color-brand-borderColor);
          --color-brand-focused-contentColor: var(--color-brand-contentColor);
          --color-brand-focused-iconColor: var(--color-brand-iconColor);
          --color-brand-focused-secondaryContentColor: var(--color-brand-focused-contentColor);
          --color-brand-hovered-background: #106ebe;
          --color-brand-hovered-borderColor: var(--color-brand-borderColor);
          --color-brand-hovered-contentColor: #ffffff;
          --color-brand-hovered-secondaryContentColor: #ffffff;
          --color-brand-iconColor: #ffffff;
          --color-brand-pressed-background: #005a9e;
          --color-brand-pressed-borderColor: var(--color-brand-borderColor);
          --color-brand-pressed-contentColor: #ffffff;
          --color-brand-pressed-iconColor: var(--color-brand-iconColor);
          --color-brand-pressed-secondaryContentColor: #ffffff;
          --color-brand-secondaryContentColor: #ffffff;
        }
        {
          -moz-osx-font-smoothing: var(--body-mozOsxFontSmoothing);
          -webkit-font-smoothing: var(--body-webkitFontSmoothing);
          background: var(--color-body-background);
          color: var(--color-body-contentColor);
          font-family: var(--body-fontFamily);
          font-size: var(--body-fontSize);
          font-weight: var(--body-fontWeight);
        }
  >
    <div>
      Dark theme
    </div>
  </div>
</div>
`;
